### PR TITLE
Im 108 revoke agreement be endpoint

### PR DESCRIPTION
**Pull request by inigo-cobian**

### CODE REVIEW

File: klab.hub/src/main/java/org/integratedmodelling/klab/hub/tasks/controllers/AgreementController.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+@RestController
+public class AgreementController {
+
+    @Autowired
+    UserAgreementService service;
+
+    @GetMapping(value = API.HUB.USER_AGREEMENTS)
+    @RolesAllowed({"ROLE_ADMINISTRATOR", "ROLE_SYSTEM"})
+    public ResponseEntity< ? > revokeAgreementOfUser(
```
Comment (iperdomo) [inline klab.hub/src/main/java/org/integratedmodelling/klab/hub/tasks/controllers/AgreementController.java:27]: Why the function is named `revokeAgreementOfUser` for a `GET` http method? 🤔

File: klab.hub/src/main/java/org/integratedmodelling/klab/hub/tasks/controllers/AgreementController.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    public ResponseEntity< ? > patchAgreementOfUser(
+            @PathVariable("id") String username,
+            @PathVariable("agreement-id") String agreementId,
+            @RequestBody Agreement agreement) {
+        if(!validateAgreementId(agreement, agreementId)) {
+            return ResponseEntity.status(HttpStatus.BAD_REQUEST)
+                    .body("Cannot modify agreement. Unclear id reference.");
+        }
+        try {
+            service.patchAgreementOfUser(username, agreement);
```
Comment (iperdomo) [inline klab.hub/src/main/java/org/integratedmodelling/klab/hub/tasks/controllers/AgreementController.java:78]: The _patch_ idea is based on the the [HTTP verb](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PATCH) (HTTP layer) right? What we're doing at the service layer is _updating_ the agreement. I would suggest to not mix layer concepts/names.

**Comment by inigo-cobian** [comment]
Applied some of the suggestions in this thread via the last commit.

**Review by iperdomo — APPROVED** [review]
👍🏾

**Pull request merged**